### PR TITLE
feat(progress-tracker): add default value for aria-valuenow

### PR DESCRIPTION
**Pull request by robinzigmond**
Automatically set aria-valuenow attribute based on the values of aria-valuemin, aria-valuemax and the progress prop. Add and improve documentation on these props to show the functionality better.

The existing tests for aria props have also been updated so that instead of just testing that the props have the values set (which doesn't really tell us anything), it tests that the appropriate HTML ARIA attributes are set on the underlying DOM node.

### Proposed behaviour

aria-valuenow should always be set on the ProgressTracker's container, even if not manually set - it should be automatically calculated based on the `progress` prop (and also the `aria-valuemin` and `aria-valuemax` in case these have custom values different from 0 and 100).

### Current behaviour

`aria-valuenow` is never set unless explicitly passed as a prop, meaning that screenreaders say "indeterminate" even when that is clearly not the developer's intention. This means that consumers have to explicitly pass in `aria-valuenow`, which is easy to forget and the omission is unlikely to even be noticed unless testing with a screenreader.

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Test ProgressTracker examples with screenreaders - particularly VoiceOver on Mac and NVDA on Windows.

### CODE REVIEW

File: src/components/progress-tracker/progress-tracker.spec.js [hunk 1]
```diff
@@ -492,20 +492,25 @@ describe("ProgressBar", () => {
...
+      ["aria-valuemin", "0"],
+      ["aria-valuenow", "10"],
+      ["aria-valuemax", "100"],
       ["aria-valuetext", "Bar"],
     ])("should allow a value to be passed to %s ", (prop, value) => {
-      expect(wrapper.prop(prop)).toBe(value);
+      expect(wrapper.getDOMNode().getAttribute(prop)).toBe(value);
+    });
+
+    it("aria-valuenow should be calculated from the values of progress, aria-valuemin and aria-valuemax if not provided", () => {
```
Comment (edleeks87) [inline src/components/progress-tracker/progress-tracker.spec.js:511]:
```suggestion
    it("should calculate aria-valuenow from the values of progress, aria-valuemin and aria-valuemax if not provided", () => {
```

**Comment by mkrds** [comment]
Nothing more to add than what Ed has pointed out

File: src/components/progress-tracker/progress-tracker.stories.mdx [hunk 2]
```diff
@@ -418,7 +418,10 @@ This component has `aria-valuemin`, `aria-valuemax`, `aria-valuenow` and `aria-v
 
 For more information on these props, please visit the W3 documentation on `role="progressbar"`. This can be accessed by clicking [here](https://www.w3.org/TR/wai-aria-1.1/#progressbar).
 
-### Example
+Note that `aria-valuenow` will by default be calculated from the provided values for `aria-valuemin` (which defaults to 0), `aria-valuemax` (which defaults to 100), and `progress`. This
+default can be over-ridden if you provide this prop explicitly.
```
[the remark this answers is not in the document]
Comment (robinzigmond) [reply src/components/progress-tracker/progress-tracker.stories.mdx:422]: seems you're right - funny that I've always hyphenated it. No big deal either way,  I'll change it.

**Review by edleeks87 — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**